includes/class-wc-stripe-blocks-support.php

```diff
@@ -0,0 +1,345 @@
+	 */
+	private function get_button_theme() {
+		return isset( $this->settings['payment_request_button_theme'] ) ? $this->settings['payment_request_button_theme'] : 'dark';
+	}
+
+	/**
+	 * Return the height for the payment button.
+	 *
+	 * @return string A pixel value for the height (defaults to '64').
+	 */
+	private function get_button_height() {
+		return isset( $this->settings['payment_request_button_height'] ) ? str_replace( 'px', '', $this->settings['payment_request_button_height'] ) : '64';
+	}
+
+	/**
+	 * Return the inline cc option.
+	 *
+	 * @return boolean True if the inline CC form option is enabled.
+	 */
+	private function get_inline_cc_form() {
+		return isset( $this->settings['inline_cc_form'] ) && 'yes' === $this->settings['inline_cc_form'];
+	}
+
+	/**
+	 * Return the locale for the payment button.
+	 *
+	 * @return string Defaults to en_US.
+	 */
+	private function get_button_locale() {
+		return apply_filters( 'wc_stripe_payment_request_button_locale', substr( get_locale(), 0, 2 ) );
+	}
+
+	/**
+	 * Return the icons urls.
+	 *
+	 * @return array Arrays of icons metadata.
+	 */
+	private function get_icons() {
+		$icons_src = [
+			'visa'       => [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/visa.svg',
+				'alt' => __( 'Visa', 'woo-gutenberg-products-block' ),
+			],
+			'amex'       => [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/amex.svg',
+				'alt' => __( 'American Express', 'woo-gutenberg-products-block' ),
+			],
+			'mastercard' => [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/mastercard.svg',
+				'alt' => __( 'Mastercard', 'woo-gutenberg-products-block' ),
+			],
+		];
+
+		if ( 'USD' === get_woocommerce_currency() ) {
+			$icons_src['discover'] = [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/discover.svg',
+				'alt' => __( 'Discover', 'woo-gutenberg-products-block' ),
+			];
+			$icons_src['jcb']      = [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/jcb.svg',
+				'alt' => __( 'JCB', 'woo-gutenberg-products-block' ),
+			];
+			$icons_src['diners']   = [
+				'src' => WC_STRIPE_PLUGIN_URL . '/assets/images/diners.svg',
+				'alt' => __( 'Diners', 'woo-gutenberg-products-block' ),
+			];
+		}
+		return $icons_src;
+	}
+
+	/**
+	 * Add payment request data to the order meta as hooked on the
+	 * woocommerce_rest_checkout_process_payment_with_context action.
+	 *
+	 * @param PaymentContext $context Holds context for the payment.
+	 * @param PaymentResult  $result  Result object for the payment.
+	 */
+	public function add_payment_request_order_meta( PaymentContext $context, PaymentResult &$result ) {
+		$data = $context->payment_data;
+		if ( ! empty( $data['payment_request_type'] ) && 'stripe' === $context->payment_method ) {
+			// phpcs:ignore WordPress.Security.NonceVerification
+			$post_data = $_POST;
+			$_POST     = $context->payment_data;
+			$this->add_order_meta( $context->order, $data['payment_request_type'] );
+			$_POST = $post_data;
```

Comment:
I think we can skip this whole dance with `$_POST` here? I don't see why we're re-assigning it, just to not use it, and then assigning the original value to it. Is there something I'm missing in `add_order_meta`?

I think this can just be reduced to

```suggestion
			$this->add_order_meta( $context->order, $data['payment_request_type'] );
```

Reply:
Have you tried it? If it works without this `$_POST` dance, then go for it.

Reply:
I'll try changing this and see what happens. Thanks for the input Darren!

Reply:
The order meta update still works correctly after removing the `$_POST` manipulation, so I'm marking this as resolved with 9eeb9d7b590f04398b4ca84795942aa926f18d2a 🙂